x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/build_events_query.ts

```diff
@@ -58,25 +58,40 @@ export const buildEventsScrollQuery = ({
       },
     },
   ];
-  return {
+  const searchQuery = {
     allowNoIndices: true,
     index,
-    scroll,
     size,
     ignoreUnavailable: true,
     body: {
       query: {
         bool: {
           filter: [
-            ...filterWithTime,
+            ...[filterWithTime],
```

Comment:
Is this doing what you expect to? It's now creating an array within an array. The older re-index API is not using this way so I'm curious which one is wrong or if both of these are equivalent?

I think this is a perfect time to create a unit test for `build_events_query.test.ts`

For example I just made this one which passes but shows a future maintainer that yes this is what you intended to do and gives a good example that the filter is returning an array within an array:


```ts
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { buildEventsSearchQuery } from './build_events_query';

describe('create_signals', () => {
  test('it builds a now-5m up to today filter', () => {
    const query = buildEventsSearchQuery({
      index: ['auditbeat-*'],
      from: 'now-5m',
      to: 'today',
      filter: {},
      size: 100,
      searchAfterSortId: undefined,
    });
    expect(query).toEqual({
      allowNoIndices: true,
      index: ['auditbeat-*'],
      size: 100,
      ignoreUnavailable: true,
      body: {
        query: {
          bool: {
            filter: [
              [
                {},
                {
                  bool: {
                    filter: [
                      {
                        bool: {
                          should: [
                            {
                              range: {
                                '@timestamp': {
                                  gte: 'now-5m',
                                },
                              },
                            },
                          ],
                          minimum_should_match: 1,
                        },
                      },
                      {
                        bool: {
                          should: [
                            {
                              range: {
                                '@timestamp': {
                                  lte: 'today',
                                },
                              },
                            },
                          ],
                          minimum_should_match: 1,
                        },
                      },
                    ],
                  },
                },
              ],
              {
                match_all: {},
              },
            ],
          },
        },
        track_total_hits: true,
        sort: [
          {
            '@timestamp': {
              order: 'asc',
            },
          },
        ],
      },
    });
  });
});
```

Reply:
lol nope - I think I messed up the spread operator outside of the array when fixing a merge conflict during a rebase with master today. I swear I didn't have the spread outside of the array before I rebased. I'm pretty sure I didn't at least..

Reply:
Just kidding I looked back and that doesn't seem to have been the case? Don't know what happened there - removing the extraneous braces to make it look normal. Thanks.